classes/core/SettingsBuilder.php

```diff
@@ -72,22 +72,33 @@ public function update(array $values)
             fn (mixed $value, string $key) => [Str::camel($key) => $value]
         );
 
-        $us = $this->model->getSettingsTable();
-        $primaryKey = $this->model->getKeyName();
-
-
-        $sql = $this->buildUpdateSql($settingValues, $us, $newQuery);
+        $schema = null;
+        if (!$this->getSchemaName()) {
+            $casts = $this->model->getCasts();
+            foreach ($this->model->getSettings() as $settingName) {
+                $schema['properties'][$settingName]['multilingual'] = in_array($settingName, $this->model->getMultilingualProps());
+
+                if (array_key_exists($settingName, $casts)) {
+                    $type = $casts[$settingName];
```

Comment:
This is where we may need to adapt a bit when a custom cast is going to apply like I have proposed in https://github.com/pkp/pkp-lib/issues/10476 . Though I am not sure without some testing .